pkg/apis/core/validation/shoot.go

```diff
@@ -207,9 +210,13 @@ func validateShootKubeconfigRotation(newMeta, oldMeta metav1.ObjectMeta, fldPath
 
 // ValidateShootSpec validates the specification of a Shoot object.
 func ValidateShootSpec(meta metav1.ObjectMeta, spec *core.ShootSpec, fldPath *field.Path, inTemplate bool) field.ErrorList {
-	allErrs := field.ErrorList{}
+	var (
+		allErrs    = field.ErrorList{}
+		workerless = len(spec.Provider.Workers) == 0
```

Comment:
With the helper function, you can get rid of "duplicating" this logic

Reply:
Helper function which accepts `ShootSpec`? We still have to define them for `v1beta1` and internal version, right?

Reply:
For example, whatever is "helping". You can also pass the whole shoot or just the list of workers.

> We still have to define them for v1beta1 and internal version, right?

Yes